torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,711 @@
+from typing import List, Optional, Callable, Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision.models.optical_flow.raft as raft
+from torch import Tensor
+from torchvision.models._api import WeightsEnum
+from torchvision.models.optical_flow._utils import make_coords_grid, grid_sample, upsample_flow
+from torchvision.models.optical_flow.raft import ResidualBlock, MotionEncoder, FlowHead
+from torchvision.ops import Conv2dNormActivation
+from torchvision.utils import _log_api_usage_once
+
+
+__all__ = (
+    "RaftStereo",
+    "raft_stereo",
+    "raft_stereo_fast",
+    "Raft_Stereo_Weights",
+    "Raft_Stereo_Fast_Weights",
+)
+
+
+def _corr_channel_normalize(corr: Tensor, num_channels: int):
+    return corr / torch.sqrt(torch.tensor(num_channels).to(corr.device))
+
+
+def _zeroes_second_channel(delta_depth: Tensor):
+    delta_depth[:, 1] = 0.0
+
+
+torch.fx.wrap("_corr_channel_normalize")
+torch.fx.wrap("_zeroes_second_channel")
+torch.fx.wrap("make_coords_grid")
+torch.fx.wrap("grid_sample")
+
+
+class BaseEncoder(raft.FeatureEncoder):
+    """Base encoder for FeatureEncoder and ContextEncoder in which weight may be shared.
+
+    See the Raft-Stereo paper section 4.6 on backbone part.
+    """
+
+    def __init__(
+        self,
+        *,
+        block: Callable[..., nn.Module] = ResidualBlock,
+        layers: Tuple[int, int, int, int] = (64, 64, 96, 128),
+        strides: Tuple[int, int, int, int] = (2, 1, 2, 2),
+        norm_layer: Callable[..., nn.Module] = nn.BatchNorm2d,
+    ):
+        # We use layers + (256,) because raft.FeatureEncoder require 5 layers
+        # but here we will set the last conv layer to identity
+        super().__init__(block=block, layers=layers + (256,), strides=strides, norm_layer=norm_layer)
+
+        # Base encoder don't have the last conv layer of feature encoder
+        self.conv = nn.Identity()
+
+        self.output_dim = layers[3]
+        num_downsampling = sum([x - 1 for x in strides])
+        self.downsampling_ratio = 2 ** (num_downsampling)
+
+
+class FeatureEncoder(nn.Module):
+    def __init__(
+        self,
+        base_encoder: BaseEncoder,
+        output_dim: int = 256,
+        shared_base: bool = False,
+        block: Callable[..., nn.Module] = ResidualBlock,
+    ):
+        super().__init__()
+        self.base_encoder = base_encoder
+        self.base_downsampling_ratio = base_encoder.downsampling_ratio
+        base_dim = base_encoder.output_dim
+
+        self.shared_base = shared_base
```

Comment:
No need to store it:
```suggestion
        self.shared_base = shared_base
```